js/main.js

```diff
@@ -1,4 +1,133 @@
-document.addEventListener('DOMContentLoaded', function() {
+{
+  const API_URL = 'https://opentdb.com/api.php?amount=10';
 
-});
+  class Quiz {
+    constructor(quizData) {
+      this.quizzes = quizData.results;
+      this.correctAnswerNum = 0;
+    }
+
+    getQuizCategory(index) {
+      return this.quizzes[index - 1].category;
+    }
+
+    getQuizDifficulty(index) {
+      return this.quizzes[index - 1].difficulty;
+    }
+  
+    getNumOfQuiz() {
+      return this.quizzes.length;
+    }
+
+    getQuizQuestion(index) {
+      return this.quizzes[index - 1].question;
+    }
+  
+    getCorrectAnswer(index) {
+      return this.quizzes[index - 1].correct_answer;
+    }
+
+    getIncorrectAnswers(index) {
+      return this.quizzes[index - 1].incorrect_answers;
+    }
+  
+    countCorrectAnswerNum(index, answer) {
+      const correctAnswer = this.quizzes[index - 1].correct_answer;
+      if(answer === correctAnswer) {
+        return this.correctAnswerNum++;
+      }
+    }
+  
+    getCorrectAnswerNum() {
+      return this.correctAnswerNum;
+    }
+  }
+
+  const titleElement = document.getElementById('head');
+  const questionElement = document.getElementById('sentence');
+  const answerContainer = document.getElementById('button__area');
+  const startButton = document.getElementById('start-button');
+  const categoryElement = document.getElementById('category');
+  const difficultyElement = document.getElementById('difficulty');
+
+  startButton.addEventListener('click', () => {
+    startButton.hidden = true;
+    fetchQuizDate(1);
```

Comment:
こちらには `await` が必要です。
```suggestion
    await fetchQuizDate(1);
```